File: project-board-server/src/model/projectTask.js
import _ from 'lodash';

class ProjectTask {
  constructor(object) {
    this.initWithObject(object);
  }

  initWithObject(object) {
    this.id = _.get(object, 'id');
    this.summary = _.get(object, 'summary');
    this.acceptanceCriteria = _.get(object, 'acceptanceCriteria');
    this.status = _.get(object, 'status');

    return this;
  }

  toJSON() {
    return { id: this.id, summary: this.summary, acceptanceCriteria: this.acceptanceCriteria, status: this.status };
  }

  hasError() {
    const errors = {};
    if (!this.summary) {
      errors.summary = 'Summary cannot be blank';
    }
    if (!this.acceptanceCriteria) {
      errors.acceptanceCriteria = 'Criteria cannot be blank';
    }
    return errors;
  }
}

export default ProjectTask;

File: projectboard/projectboard/src/main/java/com/example/projectboard/domain/ProjectTask.java
package com.example.projectboard.domain;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.validation.constraints.NotBlank;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;

@Entity
public class ProjectTask {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@NotBlank(message = "Summary cannot be blank")
	private String summary;
	@NotBlank(message="criteria cannot be blank")
	private String acceptanceCriteria;
	private String status;
	
	public ProjectTask() {
		
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getSummary() {
		return summary;
	}

	public void setSummary(String summary) {
		this.summary = summary;
	}

	public String getAcceptanceCriteria() {
		return acceptanceCriteria;
	}

	public void setAcceptanceCriteria(String acceptanceCriteria) {
		this.acceptanceCriteria = acceptanceCriteria;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}
	
}
